bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FtpFolderWatcherHandler.java

```diff
@@ -239,7 +231,7 @@ private void refreshFTPFolderInformation() {
                 if (!ftpRootDir.startsWith("/")) {
                     ftpRootDir = "/" + ftpRootDir;
                 }
-                currentFtpListing.clear();
+                List<String> currentFtpListing = new ArrayList<>();
```

Comment:
You also need to call `this.currentFtpListing = currentFtpListing;` some where later in this method, or is the currentFtpListing field no longer needed?

Reply:
Yes, this is only use in the current function, so i removed global declaration and will only keep it locally